### ISSUE TITLE
t/basic.t fails due to white space changes?

### ISSUE THREAD

**Issue by toddr** [genehack/Git-Wrapper#73]
Testing against a pretty plain vanilla git on perl 5.24.1-RC2.

I'm still investigating the exact nature of the white space differences.
```
t/basic.t .. 
# Testing git version: 2.9.2
ok 1
ok 2
ok 3 - repo is dirty
ok 4 - repo is clean
ok 5
ok 6
ok 7 - one log entry
ok 8 - id
not ok 9 - message

#   Failed test 'message'
#   at t/basic.t line 77.
#          got: 'FIRST
# 
#         BODY
# '
#     expected: 'FIRST
# 
# 	BODY
# '
ok 10 - threw Git::Wrapper::Exception
ok 11 - date
ok 12 - id
ok 13 - log(--oneline) dies
ok 14 - RUN(log --oneline) lives
ok 15 - one log entry
ok 16 - one raw log entry
ok 17
ok 18 - Attempt to commit interactively fails quickly
ok 19 - two log entries, one with empty commit message
ok 20 - using RUN('log','--format=%H') to get all 2 commit SHAs
ok 21 - argument test: long arg, no spaces in val
ok 22 - argument test: long arg, spaces in val
ok 23 - argument test: short arg, no spaces in val
ok 24 - argument test: short arg, spaces in val
ok 25 - new branch name is correct
ok 26 - passing content with -STDIN option
1..26
# Looks like you failed 1 test of 26.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/26 subtests 

Test Summary Report
-------------------
t/basic.t (Wstat: 256 Tests: 26 Failed: 1)
  Failed test:  9
  Non-zero exit status: 1
Files=1, Tests=26,  1 wallclock secs ( 0.03 usr  0.00 sys +  0.13 cusr  0.11 csys =  0.27 CPU)
Result: FAIL
```

**Comment by toddr** [comment]
I applied this patch to the test:

```diff
diff --git a/t/basic.t b/t/basic.t
index 7273859..09f51bc 100644
--- a/t/basic.t
+++ b/t/basic.t
@@ -10,6 +10,7 @@ use File::Path qw(mkpath);
 use POSIX qw(strftime);
 use Sort::Versions;
 use Test::Deep;
+use Test::HexString;
 use Test::Exception;
 
 my $dir = tempdir(CLEANUP => 1);
@@ -74,7 +75,7 @@ is(@log, 1, 'one log entry');
 
 my $log = $log[0];
 is($log->id, (split /\s/, $rev_list[0])[0], 'id');
-is($log->message, "FIRST\n\n\tBODY\n", "message");
+is_hexstr($log->message, "FIRST\n\n\tBODY\n", "message");
 
 throws_ok { $git->log( "--format=%H" ) } q{Git::Wrapper::Exception};
```

This was the output:
```
not ok 9 - message

#   Failed test 'message'
#   at t/basic.t line 79.
#   at bytes 0-0xf (0-15)
#   got: | 46 49 52 53 54 0a 0a 20 20 20 20 20 20 20 20 42 |FIRST..        B|
#   exp: | 46 49 52 53 54 0a 0a 09 42 4f 44 59 0a .. .. .. |FIRST...BODY.   |
```

It looks like the output format changed from spaces to tabs for some reason. We're in the middle of a re-build while moving to 5.24 so git has probably been this way for a while. We've not noticed any ill effects so probably this is just a unit test that needs to be switched to like?

**Comment by toddr** [comment]
oops missed it. Looks like pull request #72 should fix this.

**Comment by genehack** [comment]
I just merged #77, which should fix this issue. Dev release coming once I clear out the rest of the queue. 

Closing this ticket now, but please re-open if the dev release doesn't resolve the issue.
